nailgun/entities.py

```diff
@@ -460,6 +460,69 @@ def update(self, fields=None):
         return self.read()
 
 
+class ArfReport(
+        Entity,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin):
+    """A representation of a Arf Report entity.
+
+    # Read Arf report
+    ArfReport(id=<id>).read()
+    # Delete Arf report
+    ArfReport(id=<id>).delete()
+    # Download Arf report in HTML
+    ArfReport(id=<id>).download_html()
+    """
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'location': entity_fields.OneToManyField(Location),
+            'organization': entity_fields.OneToManyField(Organization),
+            'host': entity_fields.OneToOneField(Host),
+            'openscap_proxy': entity_fields.OneToOneField(Capsule),
+            'policy': entity_fields.OneToOneField(CompliancePolicies)
+        }
+        self._meta = {
+            'api_path': 'api/compliance/arf_reports',
+        }
+        super(ArfReport, self).__init__(server_config, **kwargs)
+
+    def path(self, which=None):
+        """Extend ``nailgun.entity_mixins.Entity.path``.
+        The format of the returned path depends on the value of ``which``:
+
+        download_html
+            /api/compliance/arf_reports/:id/download_html
+
+        Otherwise, call ``super``.
+
+        """
+        if which in ('download_html',):
```

Comment:
what @mirzal says anyone would do

Reply:
@mirzal This will be used in the scenario where path can be different but here it is specific.

Reply:
@mirzal had a talk with nikhil, It seems we are comparing which with download_html and then creating the path.

Reply:
Then can't we use normal comparison, `if which == 'download_html'`?

I don't understand why this is done in non-obvious way. Is there any benefit of checking if one-element tuple contains string from variable, instead of just checking if string in variable is the same as string we compare against?

Reply:
@mirzal yes we can, generally we keep tuple to extend other path definitions.

Reply:
@ntkathole I can see this point. Thanks.

Personally, I would use `==` and switch to tuple only when we actually have multiple paths to handle. I find it easier to understand. But we use both ways in Nailgun, so we can leave it as is.